src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -271,24 +256,58 @@ EnsureModificationsCanRun(void)
 
 
 /*
- * IsCitusTable returns whether relationId is a distributed relation or
- * not.
+ * DecrementCacheEntryRefCount decrements the cache entry's refCount,
+ * & frees the entry if it is no longer referenced.
+ */
+static void
+DecrementCacheEntryRefCount(CitusTableCacheEntry *cacheEntry)
+{
+	Assert(cacheEntry->refCount > 0);
+	cacheEntry->refCount--;
+	if (cacheEntry->refCount == 0)
+	{
+		ResetCitusTableCacheEntry(cacheEntry);
+		pfree(cacheEntry);
+	}
+}
+
+
+/*
+ * ReleaseTableCacheEntry releases a pointer loaned out by cache.
+ * This only asserts correctness, without asserts this is a nop.
```

Comment:
I don't get the assertion part. Where are the asserts?

Reply:
Stale comment from when this was only asserts